### PR TITLE
Add reference documentation

### PR DESCRIPTION
**Pull request by fpagnoux**
Available on http://openfisca.readthedocs.io/.

### CODE REVIEW

File: doc/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+openfisca_core >= 4.2.0, < 5.0
```
Comment (fpagnoux) [inline doc/requirements.txt]: As I mentioned in the original PR, pip is really bad at managing dependencies with a relative path inside a `requirements.txt` file.

If I put `..` instead of `openfisca_core >= 4.2.0, < 5.0`, it will look for a `setup.py` in the parent directory relatively to _where the instruction is executed from_, and not _where the requirements file is_.

File: doc/REAMDE.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Openfisca-Core reference documentation
+
+Source of the [openfisca-core reference documentation](http://openfisca.readthedocs.io/).
+
+## Installing
+
+```sh
+    pip install -r requirements.txt
+    pip install sphinx-autobuild
```
Comment (MattiSG) [inline doc/REAMDE.md]: > Either indent or use the triple-backtick syntax to trigger code blocks. I recommend triple-backticks over indentation, since it allows specifying the language.

File: doc/REAMDE.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Openfisca-Core reference documentation
+
+Source of the [openfisca-core reference documentation](http://openfisca.readthedocs.io/).
+
+## Installing
+
+```sh
+    pip install -r requirements.txt
```
Comment (MattiSG) [inline doc/REAMDE.md]: > Either indent or use the triple-backtick syntax to trigger code blocks. I recommend triple-backticks over indentation, since it allows specifying the language.

File: doc/REAMDE.md (hunk 2 above)
Comment (MattiSG) [inline doc/REAMDE.md]: Why isn't this in the requirements file?

File: doc/requirements.txt (hunk 1 above)
Comment (MattiSG) [inline doc/requirements.txt]: Add this piece of information as a comment in the file itself, otherwise someone else (including yourself in 6 months) will try to change it again.

File: doc/REAMDE.md (hunk 3 above)
Comment (MattiSG) [inline doc/REAMDE.md]: If this is intended to be run in the `doc` folder, it is missing a `cd`.

File: doc/REAMDE.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Openfisca-Core reference documentation
```
Comment (MattiSG) [inline doc/REAMDE.md]: Typo in filename.

File: doc/REAMDE.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+Source of the [openfisca-core reference documentation](http://openfisca.readthedocs.io/).
```
Comment (MattiSG) [inline doc/REAMDE.md]: Inconsistent casing of `Openfisca-Core`.

Comment (MattiSG) [inline doc/REAMDE.md]: Use HTTPS.

File: doc/REAMDE.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+```sh
+    pip install -r requirements.txt
+    pip install sphinx-autobuild
+```
+
+## Usage
+
+```sh
+    sphinx-autobuild source build
```
Comment (MattiSG) [inline doc/REAMDE.md]:
```
› pip install sphinx-autobuild --user
Collecting sphinx-autobuild
[…]
Installing collected packages: pathtools, argh, watchdog, port-for, singledispatch, certifi, backports-abc, tornado, livereload, sphinx-autobuild
Successfully installed argh-0.26.2 backports-abc-0.5 certifi-2017.1.23 livereload-2.5.1 pathtools-0.1.2 port-for-0.3.1 singledispatch-3.4.0.3 sphinx-autobuild-0.6.0 tornado-4.4.2 watchdog-0.8.3
› sphinx-autobuild source build
-bash: sphinx-autobuild: command not found
```

File: doc/source/openfisca-run-test.rst [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+.. code-block:: shell
+
+  openfisca-run-test test_3.yaml
+  # Failure: the third test does not pass, as its error margin is by default 0
+
+  openfisca-run-test -m 0.01 test_3.yaml
+  # Success:
+  # The first test passes with its own 0.05 relative error margin
+  # The second test passes with its own 10 absolute error margin
+  # The third test passes with the default 0.01 relative error margin
```
Comment (MattiSG) [inline doc/source/openfisca-run-test.rst]: This could benefit from a short theoretical explanation of what is an error margin and why it is useful… Might not be part of this PR though.

File: doc/source/test_runner.rst [hunk 8]
```diff
@@ -0,0 +1,9 @@
+=================
+tools.test_runner
+=================
+
+.. automodule:: openfisca_core.tools.test_runner
+    :members:
+
+
+Tests can also be run in command line with :any:`openfisca-run-test`.
```
Comment (MattiSG) [inline doc/source/test_runner.rst]: “in the console” or “on the command line”.

File: doc/REAMDE.md (hunk 2 above)
Comment (fpagnoux) [inline doc/REAMDE.md]: `sphinx-autobuild` is the most convenient tool to build the doc, but it is not strictly necessary (for instance `readthedocs` doesn't use it).
I could put it in an `optional-requirements.txt` file, would that be better ?

**Comment by MattiSG** [comment]
I don’t know what’s the convention with Python. Maybe there’s a notion of “dependency group” that can be triggered with an option?

File: doc/REAMDE.md (hunk 6 above)
Comment (fpagnoux) [inline doc/REAMDE.md]: 2017, and some people still not use virtual environments :trollface: ?

Comment (fpagnoux) [inline doc/REAMDE.md]: I've added a recommandation to use a virtualenv in the README

**Comment by fpagnoux** [comment]
This exists for proper packages (with a `setup.py` file), but not when you just use a `requirements.txt`. I'm not sure this is worth making the doc its own package.

**Comment by fpagnoux** [comment]
@MattiSG is this GTM ?

File: doc/source/openfisca-run-test.rst [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+  # Success
+
+  openfisca-run-test test.yaml
+  # Success: the country package is automatically detected.
+  # May fail if several country packages are installed in your environment.
+  # In that case, specify which package to use with the --country_package option
+
+
+Force option
+^^^^^^^^^^^
```
Comment (MattiSG) [inline doc/source/openfisca-run-test.rst]: `openfisca-run-test.rst:43: WARNING: Title underline too short.`

**Comment by MattiSG** [comment]
Needs a version bump. I'm in favour of a minor version bump.

**Comment by fpagnoux** [comment]
Agreed, but I don't think this should be lock this PR. I've opened https://github.com/openfisca/openfisca-core/issues/441.

**Comment by MattiSG** [comment]
> I don't think this should lock this PR

Sure, hence the ✅`Accept` 😄 

Good job!!
